x-pack/plugins/aiops/public/hooks/use_data_source.tsx

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { createContext, FC, useCallback, useContext, useEffect, useState } from 'react';
+import { DataView } from '@kbn/data-views-plugin/common';
+import type { SavedSearch } from '@kbn/saved-search-plugin/public';
+import { EuiEmptyPrompt } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useAiopsAppContext } from './use_aiops_app_context';
+
+export const DataSourceContext = createContext<DataViewAndSavedSearch>({
+  get dataView(): never {
+    throw new Error('DataSourceContext is not implemented');
+  },
+  savedSearch: null,
+});
+
+export function useDataSource() {
+  return useContext(DataSourceContext);
+}
+
+export interface DataViewAndSavedSearch {
+  savedSearch: SavedSearch | null;
+  dataView: DataView;
+}
+
+/**
+ * Context provider that resolves current data view and the saved search
+ *
+ * @param children
+ * @constructor
+ */
+export const DataSourceContextProvider: FC<{ dataViewId?: string; savedSearchId?: string }> = ({
```

Comment:
I think we have a very similar hook/context across ML and Data Visualizer... Would be useful to have something like this is a common package 🤔

Reply:
As discussed in Slack, I agree this is good to do in a follow up 👍